sympy/physics/mechanics/joint.py

```diff
@@ -474,6 +474,50 @@ def _locate_joint_frame(self, body, interframe):
         body.masscenter.set_vel(interframe, 0)  # Fixate interframe to body
         return interframe
 
+    def _fill_coordinate_list(self, coordinates, n_coords, label='q', offset=0):
+        """Helper method for _generate_coordinates and _generate_speeds.
+
+        Parameters
+        ==========
+
+        coordinates : List
+            List of coordinates or speeds that have been provided.
+        n_coords : Integer
+            Number of coordinates that should be returned.
+        label : String, optional
+            Coordinate type either 'q' (coordinates) or 'u' (speeds).
+        offset : Integer
+            Count offset when creating new dynamicsymbols.
+
+        """
+        name = 'generalized coordinate' if label == 'q' else 'generalized speed'
```

Comment:
As this is only needed to help construct error messages, I'd recommend just having it (two copies) directly before each of the two potential `ValueError` exceptions are raised. That way it'd avoid the variable assignment if it is not required.

Reply:
This way it still fitted on a single line, otherwise its twice 4 extra lines.

Reply:
Having seen the changed version I think I now agree with you here that it was neater before.